rotkehlchen/chain/ethereum/modules/convex/decoder.py

```diff
@@ -34,23 +41,50 @@
 from rotkehlchen.constants.assets import A_CRV, A_CVX
 from rotkehlchen.errors.asset import UnknownAsset, WrongAssetType
 from rotkehlchen.logging import RotkehlchenLogsAdapter
-from rotkehlchen.types import CURVE_POOL_PROTOCOL, ChecksumEvmAddress
+from rotkehlchen.types import CURVE_POOL_PROTOCOL, CacheType, ChecksumEvmAddress
 from rotkehlchen.utils.misc import hex_or_bytes_to_address, hex_or_bytes_to_int
 
+if TYPE_CHECKING:
+    from rotkehlchen.chain.ethereum.node_inquirer import EthereumInquirer
+    from rotkehlchen.chain.evm.decoding.base import BaseDecoderTools
+    from rotkehlchen.user_messages import MessagesAggregator
+
 logger = logging.getLogger(__name__)
 log = RotkehlchenLogsAdapter(logger)
 
 DEPOSIT_EVENT = b'\x9eq\xbc\x8e\xea\x02\xa69i\xf5\t\x81\x8f-\xaf\xb9%E2\x90C\x19\xf9\xdb\xday\xb6{\xd3J_='  # noqa: E501
 REWARD_ADDED = b'\xde\x88\xa9"\xe0\xd3\xb8\x8b$\xe9b>\xfe\xb4d\x91\x9ck\xf9\xf6hW\xa6^+\xfc\xf2\xce\x87\xa9C='  # noqa: E501
 
 
-class ConvexDecoder(DecoderInterface):
-
-    def _decode_convex_events(self, context: DecoderContext) -> DecodingOutput:
+class ConvexDecoder(DecoderInterface, ReloadablePoolsAndGaugesDecoderMixin):
+    def __init__(
+            self,
+            ethereum_inquirer: 'EthereumInquirer',
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator',
+    ) -> None:
+        super().__init__(
+            evm_inquirer=ethereum_inquirer,
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+        )
+        ReloadablePoolsAndGaugesDecoderMixin.__init__(
+            self,
+            evm_inquirer=ethereum_inquirer,
+            cache_type_to_check_for_freshness=CacheType.CONVEX_POOL_ADDRESS,
+            query_data_method=query_convex_data,
+            save_data_to_cache_method=save_convex_data_to_cache,
+            read_pools_and_gauges_from_cache_method=read_convex_data_from_cache,
+        )
```

Comment:
since you only need pools here... wouldn't it make sense to split `ReloadablePoolsAndGaugesDecoderMixin` into `ReloadablePoolsDecoderMixin` and `ReloadablePoolsAndGaugesDecoderMixin (ReloadablePoolsDecoderMixin)` a use the pool only one here?